db/versions/0009.yml

```diff
@@ -212,3 +212,92 @@ methods:
+    serviceName: purge_cache
+    args: provisioner_id_in text, worker_type_in text
+    returns: table (provisioner_id text, worker_type text, cache_name text, before timestamptz)
+    body: |-
+      begin
+        return query
+        select cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name, cache_purges.before from cache_purges
+        where cache_purges.provisioner_id = purge_requests.provisioner_id_in and cache_purges.worker_type = purge_requests.worker_type_in;
+      end
+  all_purge_requests:
+    description: |-
+      View all active purge requests.
+      This is useful mostly for administors to view
+      the set of open purge requests. It should not
+      be used by workers. They should use the purgeRequests
+      endpoint that is specific to their workerType and
+      provisionerId.
+    mode: read
+    serviceName: purge_cache
+    args: page_size_in integer, page_offset_in integer
+    returns: table (provisioner_id text, worker_type text, cache_name text, before timestamptz)
+    body: |-
+      declare
+      sql text := 'select cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name, cache_purges.before from cache_purges';
+      begin
+        return query select cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name, cache_purges.before
+        from cache_purges
+        order by cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name
+        limit case
+          when (page_size_in is not null and page_size_in > 0) then page_size_in
+          else null
+        end
+        offset case
+          when (page_offset_in is not null and page_offset_in > 0) then page_offset_in
+          else 0
+        end;
+      end
+  purge_cache:
+    serviceName: purge_cache
+    description: |-
+      Publish a request to purge caches named `cacheName` with
+      on `provisionerId`/`workerType` workers.
+      If such a request already exists, its `before` timestamp is updated to
+      the current time.
+    mode: write
+    args: prov_id text, wt text, c_name text, bef timestamptz, exp timestamptz
+    returns: void
+    body: |-
+      declare
+        new_etag uuid := public.gen_random_uuid();
+      begin
+        insert into cache_purges(provisioner_id, worker_type, cache_name, before, expires, etag)
+        values (
+          prov_id,
+          wt,
+          c_name,
+          bef,
+          exp,
+          new_etag
+        ) on conflict (provisioner_id, worker_type, cache_name) do
+        update
+        set (before, expires, etag) = (purge_cache.bef, exp, new_etag)
+        where cache_purges.provisioner_id = purge_cache.prov_id and cache_purges.worker_type = purge_cache.wt and cache_purges.cache_name = purge_cache.c_name;
+      end
+  cache_purges_expires:
+    description: |-
+      Expire cachePurges that are past their expiration.
+      Returns a promise that all expired cachePurges have been deleted.
```

Comment:
This isn't JS, it doesn't return a promise ;)

Generally, the descriptions here need a little bit of tweaking to describe the DB functions, rather than the API functions that call them.